src/libraries/System.Reflection/overview.md

```diff
@@ -0,0 +1,37 @@
+- **Static introspection** through [System.Reflection.MetadataLoadContext](../System.Reflection.MetadataLoadContext\readme.md). This exposes the same reflection types as runtime introspection, but they are loaded into a "metadata context" where the types returned for introspection do not support invocation, and they are different instances from the types returned from runtime introspection. Because of this, this allows any assembly to be read, even if it's for a different platform provided that all assembly references are accounted for.  
+- **Runtime generation of IL for invocation** through [System.Reflection.Emit](../System.Reflection.Emit\readme.md). This is used to generate types dynamically which are commonly used to generate proxies and other dynamic classes that may not be known ahead-of-time (in which case C# generation could be used, for example). It is also used to invoke members more efficiently than the [`standard runtime invoke mechanism`](https://learn.microsoft.com/dotnet/api/system.reflection.methodbase.invoke). However,  due to recent work in 7.0 (which is to be extended in 8.0), this IL generation is now done automatically so the need to manually generate IL for performance is becoming less useful.
+- **Low-level reading and writing of assemblies**  through [System.Reflection.Metadata](../System.Reflection.Metadata\readme.md). This is used for tooling and compiler-like scenarios.
+- Other reflection libraries that extend the functionality above or serve as a container assembly to support NetStandard or attribute types emitted by compilers:
+  - [System.Reflection.Context](../System.Reflection.Context/readme.md)
+  - [System.Reflection.DispatchProxy](../System.Reflection.DispatchProxy/readme.md)
+  - [System.Reflection.Extensions](../System.Reflection.Extensions/readme.md)
+  - [System.Reflection.Primitives](../System.Reflection.Primitives/readme.md)
+  - [System.Reflection.TypeExtensions](../System.Reflection.TypeExtensions/readme.md)
+
+## <a name="status"></a>Status descriptions
+Each library has its own `readme.md` that contains a description plus  a current status of:
+- **Legacy**  
```

Comment:
@jeffhandley I thought you have a plan to do similar labelling for all packages. Does this align with your plan?